doc/api/packages.md

```diff
@@ -357,6 +377,40 @@ import submodule from 'es-module-package/private-module.js';
 // Throws ERR_PACKAGE_PATH_NOT_EXPORTED
 ```
 
+Even though subpaths provide an aribtrary string mapping to the package
+interface, it is recommended (but not required) to use explicit file extensions
+when defining package subpaths so that package consumers write
+`import 'pkg/subpath.js'` instead of `import 'pkg/subpath'` as this simplifies
+interoperability with [import maps][] and also mirrors the requirement of using
```

Comment:
i don't think this is an accurate claim. Compatibility with import maps is identical with or without extensions.

Reply:
I've added a new dedicated section on import maps compatibility.